packages/itmat-ui-react/src/components/lxd/lxd.instance.console.tsx

```diff
@@ -0,0 +1,177 @@
+                },
+                onsuccess: () => {
+                    handleResize();
+                },
+                onagent: handleResize
+            });
+
+            // // Clipboard from client to server
+            // document.addEventListener('copy', (e) => {
+            //     const clipboardData = e.clipboardData;
+            //     if (clipboardData) {
+            //         const text = clipboardData.getData('text/plain');
+            //         // If window.spice_connection is potentially undefined, check if it's defined before using it
+            //         window.spice_connection?.agent.clipboard_grab(text, SpiceMainConn.TYPE_TEXT);
+            //     }
+            //     e.preventDefault(); // Prevent the default copy action
+            // });
+
+            // // Clipboard from server to client
+            // window.spice_connection.add_channel_event_listener('spice_channel', 'agent', (agent, type) => {
+            //     if (type === SpiceMainConn.TYPE_CLIPBOARD) {
+            //         agent.clipboard_request(SpiceMainConn.TYPE_TEXT);
+            //     } else if (type === SpiceMainConn.TYPE_CLIPBOARD_SELECTION) {
+            //         // Handle clipboard selection if needed
+            //     }
+            // });
+
+            // window.spice_connection.agent.addEventListener('message', (e) => {
+            //     if (e.type === 'clipboard') {
+            //         const text = e.data; // The text from the Spice server's clipboard
+            //         navigator.clipboard.writeText(text).then(() => {
+            //             console.log('Clipboard text copied to client.');
+            //         }).catch((err) => {
+            //             console.error('Failed to copy text to clipboard:', err);
+            //         });
+            //     }
+            // });
+        } catch (e) {
+            console.error('Spice create error', e);
+        }
+
+
+        return control;
+    }, [instanceName]);
+
+
+    useEffect(() => {
+
+        // TODO !!!
+        // Should be removed and use proper async traps
+        if (hasStarted || (window as any).hasSpice)
+            return;
+
+        (window as any).hasSpice = true;
+
+        setHasStarted(true);
+        const controlWebsocketPromise = openVgaConsole();
+        return () => {
+            try {
+                window.spice_connection?.stop();
+            } catch (e) {
+                console.error('Connection opening failed', e);
+            }
+            void controlWebsocketPromise.then((controlWebsocket) => {
+                controlWebsocket?.close();
+            });
+        };
+    }, [hasStarted, openVgaConsole]);
+
+    useEffect(() => {
+        updateVgaConsoleSize();
+        window.addEventListener('resize', updateVgaConsoleSize);
+
+        return () => {
+            window.removeEventListener('resize', updateVgaConsoleSize);
+        };
+    }, []);
+
```

Comment:
For fixing the window auto resize problem, you can add `useEventListener("resize", handleResize); ` here. (You need to install this package and `import useEventListener from "@use-it/event-listener";` at the top). This will listen resize event and handle the resize.